core/tests/test_utils.py

```diff
@@ -1251,17 +1276,40 @@ def assertRaises(self, *args, **kwargs):
             'self.assertRaises should not be used in these tests. Please use '
             'self.assertRaisesRegexp instead.')
 
-    def assertRaisesRegexp(  # pylint: disable=keyword-arg-before-vararg
-            self, expected_exception, expected_regexp, callable_obj=None,
-            *args, **kwargs):
-        if not expected_regexp:
+    def assertRaisesRegexp(  # pylint: disable=invalid-name
+            self, expected_exception, expected_regex, *args, **kwargs):
+        """Asserts that the message in a raised exception matches a regex.
+        This is a wrapper around assertRaisesRegex in unittest that enforces
+        strong regex.
+
+        Args:
+            expected_exception: Exception. Exception class expected
+                to be raised.
+            expected_regex: re.Pattern|str. Regex expected to be found in
+                error message.
+            *args: list(*). Function to be called and extra positional args.
```

Comment:
If there'll always be a function, why not separate the function into its own named arg?

Reply:
There doesn't have be a function (in the case when this method is used together with `with`. And since this is inherited funciton, we cannot change the signature.

Reply:
But the old version had callable_obj, which this one doesn't. What was that for -- is that no longer needed?

Reply:
Yes, it is no longer needed.

It was used for cases where we didn't use it with `with`, you could do something like `assertRaisesRegexp(Exception, 'Somthing', callable_obj=function_to_call)` and it tested if `function_to_call` raised `Exception`. It was used sparsly in our codebase and I decided to go just with using it with `with` and removed this usecase.